packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -434,18 +440,26 @@ class ScheduleManagerCore {
                     });
             }
         }
+
         if (batchMetadata) {
             assert(this.currentBatchClientId === undefined, "there can't be active batch");
+            assert(!this.localPaused, "we should be processing ops when there is no active batch");
             this.pauseSequenceNumber = message.sequenceNumber;
             this.currentBatchClientId = message.clientId;
-            this.setPaused(true);
+            // Start of the batch
+            // Only pause processing if queue has no other ops!
+            // If there are any other ops in the queue, processing will be stopped when they are processed!
+            if (this.deltaManager.inbound.length === 1) {
+                this.setPaused(true);
+            }
         } else if (batchMetadata === false) {
-            assert(this.localPaused, "active batch, should be paused");
             this.pauseSequenceNumber = undefined;
```

Comment:
We should assert here that currentBatchClientId and pauseSequenceNumber are not undefined since we should be processing a batch.

Reply:
It is done above implicitly:
(line 432) batchMetadata === false => this.currentBatchClientId === message.clientId => not undefined
line (409) this.currentBatchClientId != undefined => this.pauseSequenceNumber != undefined

But yes, I was forced by compiler to add extra assert here in my next PR (that follows that :)) as I needed to touch this.pauseSequenceNumber and compiler does not understand it can't be undefined here. So I'll add it in future PR

Reply:
sounds good.